src/Compilers/Core/Portable/Compilation.EmitStream.cs

```diff
@@ -72,14 +72,33 @@ internal Func<Stream> GetCreateStreamFunc(DiagnosticBag diagnostics)
                 return () => CreateStream(diagnostics);
             }
 
-            internal void Dispose()
+            internal void Close()
```

Comment:
Thinking about this a little more, I think we should rename `EmitStream` to something else to make it clear that it doesn't implement `Stream`. Maybe `EmitStreamWrapper` or something?